base/array.jl

```diff
@@ -2210,7 +2210,7 @@ end
 
 function union(vs...)
     ret = Vector{promote_eltype(vs...)}()
-    seen = Set()
+    seen = Set{Any}()
```

Comment:
If I'm not mistaken, it seems like in this case we could do better than `Any`, e.g.
```julia
T = promote_eltype(vs...)
ret = Vector{T}()
seen = Set{T}()
...
```

Reply:
I did this change in a pending [PR](https://github.com/JuliaLang/julia/pull/23528/files#diff-d39084ec0fef2ad65e4b5fe0fcfaab8eR106), so if it gets accepted (I hope!) maybe no need to bother here